src/LineaStateBridge.sol

```diff
@@ -1,17 +1,251 @@
+                || _crossDomainMessenger == address(0)
+        ) {
+            revert AddressZero();
+        }
+
+        lineaWorldIDAddress = _lineaWorldIDAddress;
+        worldIDAddress = _worldIDIdentityManager;
+        crossDomainMessengerAddress = _crossDomainMessenger;
+        _gasLimitPropagateRoot = DEFAULT_LINEA_GAS_LIMIT;
+        _gasLimitSetRootHistoryExpiry = DEFAULT_LINEA_GAS_LIMIT;
+        _gasLimitTransferOwnership = DEFAULT_LINEA_GAS_LIMIT;
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                          PUBLIC API                         ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice Sends the latest WorldID Identity Manager root to the ILineaStack.
+    /// @dev Calls this method on the L1 Proxy contract to relay roots to the destination Linea Stack chain
+    function propagateRoot() external {
+        uint256 latestRoot = IWorldIDIdentityManager(worldIDAddress).latestRoot();
+
+        // The `encodeCall` function is strongly typed, so this checks that we are passing the
+        // correct data to the optimism bridge.
+        bytes memory message = abi.encodeCall(ILineaWorldID.receiveRoot, (latestRoot));
+
+        IMessageService(crossDomainMessengerAddress).sendMessage(
+            // Contract address on the Linea Stack Chain
+            lineaWorldIDAddress,
+            _gasLimitPropagateRoot,
+            message
+        );
+
+        emit RootPropagated(latestRoot);
+    }
+
+    /// @notice Adds functionality to the StateBridge to transfer ownership
+    /// of LineaWorldID to another contract on L1 or to a local Linea Stack chain EOA
+    /// @dev  _messageService should be hardcoded to 0x508Ca82Df566dCD1B0DE8296e70a96332cD644ec for L2 Linea
+    /// @dev _remoteSender should be address of L1 LineaStateBridge
+    /// @custom:revert if _owner is set to the zero address
+    // function transferOwnershipLineas(address _messageService, address _remoteSender) external onlyOwner {
+    //     if (_remoteSender == address(0) || _messageService == address(0)) {
+    //         revert AddressZero();
+    //     }
+
+    //     // The `encodeCall` function is strongly typed, so this checks that we are passing the
+    //     // correct data to the Linea Stack chain bridge.
+    //     bytes memory message =
+    //         abi.encodeCall(MessageServiceBase.updateMessageServiceBase(_messageService, _remoteSender));
+
+    //     IMessageService(crossDomainMessengerAddress).sendMessage(
+    //         // Contract address on the Linea Stack Chain
+    //         lineaWorldIDAddress,
+    //         _gasLimitTransferOwnership,
+    //         message
+    //     );
+
+    //     emit OwnershipTransferredLinea(owner(), _messageService, _remoteSender);
+    // }
+    function transferOwnershipLineas(address _messageService, address _remoteSender) external onlyOwner {
```

Comment:
Could you please explain the naming? Does the last `s` have any special meaning?

Reply:
nope there isn't any special meaning, will remove the `s`